src/transformers/file_utils.py

```diff
@@ -205,6 +206,20 @@ def filename_to_url(filename, cache_dir=None):
     return url, etag
 
 
+def path_to_str(path_like_or_str):
+    """
+    Attempts to convert a path-like object to a string.
+    """
+    if hasattr(path_like_or_str, "__fspath__"):
+        return path_like_or_str.__fspath__()
+    elif isinstance(path_like_or_str, pathlib.Path):
+        return str(path_like_or_str)
+    elif isinstance(path_like_or_str, str):
+        return path_like_or_str
+    else:
+        raise ValueError("path_like_or_str is not a path-like object or string.")
```

Comment:
Is this the implementation from pandas?

Reply:
Mostly, I also changed the default behaviour to a `ValueError` if it's not a supported type (path-like or str). I also changed some of the names to fall in line with the naming of vars/funcs in transformers.